### DISCUSSION TITLE
db = Deta().Base('log_name') seems to require environment variables to be set.

### DISCUSSION THREAD

**Discussion by xu2xulim** [deta/cloud-docs#310] [Help] [answered]
Previously I had in my main.py

db = Deta(os.environ.get('DETA_PROJECT_KEY')).Base(os.environ.get('BASE_LOG'))
and have change it to
db = Deta().Base('log_name') and deployed
Micro is running in default just like many other micros

Today when I tried the endpoint, it failed at db.put(...etc). After trying empty .env and running deta deploy -e .env,
 deta details shows that NO environment variables but it still failed.

I restored the .env but without changing the definition of db and the program works.
Is there an explanation for this and what is the right corrections steps to be taken. Thanks

**Comment by cofob** [comment]
Use `Deta(os.environ.get('DETA_PROJECT_KEY')).Base(os.environ.get('BASE_LOG'))` ([Example Base usage](https://docs.deta.sh/docs/base/py_tutorial#setup)).

First you need to initialize Deta class with your access token (so you can't just call bare `Deta()`), in Micro your token is pre-set in [DETA_PROJECT_KEY](https://docs.deta.sh/docs/micros/env_vars#pre-set-environment-variables), so you can just call in you code `deta = Deta(os.environ.get('DETA_PROJECT_KEY'))`. After that you can create Base class instance from your Deta object.

**Reply by xu2xulim** [reply]
My understanding is different. I had many micros using base that I never need to do that. This is taken from Deta 

>>from Deta
NOTE
If you are using Deta Base within a [Deta Micro](https://docs.deta.sh/docs/micros/about), the Deta SDK comes pre-installed and a valid project key is pre-set in the Micro's environment. There is no need to install the SDK or pass a key in the the initialization step.

from deta import Deta

deta = Deta()
>> end Deta

It seems that once you start with using a get from environment variables you cannot revert to use Deta() or you simply need to ensure DETA_PROJECT_ID is there...you cannot update the .env since it will erase the pre-installed.

**Reply by abdelhai** [reply]
no need to explicitly grab the project key we automatically load it if we find it. micros always have a project key preloaded excpet when they get deleted/overridden by mistake. thus `db = Deta().Base('log_name')` is valid and should always work. 

locally, you could set the project key with the default name and we also should be able to detect it.

**Comment by abdelhai** [comment, marked as answer]
I believe you did override your deta project key `DETA_PROJECT_KEY` by mistake. You should be able to fix that by setting it again yourself (`deta update -e .env`). You can get a new project key from our UI.

**Reply by xu2xulim** [reply]
It is probably the case as you have described. I remember that generated a new project ID when I realised I had accidentally pushed the .env to Github. I will marked this as answered. Thanks
